src/Access/ContextAccess.cpp

```diff
@@ -385,6 +385,9 @@ bool ContextAccess::checkAccessImplHelper(const AccessFlags & flags, const Args
     if (!getUser())
         return access_denied("User has been dropped", ErrorCodes::UNKNOWN_USER);
 
+    if (flags == AccessType::CLUSTER && !access_control->doesOnClusterQueriesRequireClusterGrant())
+        return access_granted();
```

Comment:
`flags` is a bitmask and if `doesOnClusterQueriesRequireClusterGrant()` we need to ignore some bits and then check other bits as usual

Reply:
Nice catch! Fixed.